### DISCUSSION TITLE
[Done] Packaging with JARs

### DISCUSSION THREAD

**Discussion by lokka30** [ArcanePlugins/Treasury#103] [Q&A] [answered]
Considering Treasury will support multiple platforms in the future: will there be a single jar to download that runs on all server impls? or a file for each impl?

**Comment by lokka30** [comment]
Tagging those who may be interested: @MrIvanPlays @Jikoo @MrNemo64

**Comment by MrIvanPlays** [comment, marked as answer]
Multiple jars are less prone to errors. Im for multiple jars.

**Comment by lokka30** [comment]
How will we distribute these JARs eventually?

* LuckPerms for instance uses its own website for all non-Bukkit builds, e.g. bungee, velocity.
* We can upload multiple JARs for Releases on GitHub :) so we can distribute all non-Bukkit builds there
* Don't like this option .. but we can zip all of the jar files and just upload the zip to spigot. One file download, contains multiple jars inside it. Can confuse some users and be annoying for everyone
* Other options?...

**Reply by MrIvanPlays** [reply]
spigot will only distribute the spigot version, and if anyone else wants bungee, velocity etc. can come to github or wherever we host all the jars.

**Reply by lokka30** [reply]
perhaps we can also route spigot download button to github

**Reply by Jikoo** [reply]
It could still be an external link to the file in the GH release so it doesn't have to be uploaded (or built) manually, but I think a direct Spigot version download would be more convenient on Spigot - it would reduce accidental downloads and extra clicks.

**Reply by MrNemo64** [reply]
By routing the spigot download button to github we disable the posibility of using Treasury on Aternos as it only allows to use plugins that have a direct link on spigotmc.org
I don't know if is something to worry about but to mention it

**Reply by lokka30** [reply]
Definitely is something to worry about, don't want to lose Aternos users. That's fine, we can just link to the github release of each new version so people can see the other JARs, spigot will have the Treasury-Bukkit download.
